File: 5/05525.cpp
#include <iostream>
using namespace std;

int PN(string s, int n) {
	int cnt = 0;
	int num = 0;

	for (int i = 0; i < s.length() - 2; i++) {
		if (s[i] == 'I' && s[i + 1] == 'O' && s[i + 2] == 'I') {
			num++;	// Pn 하나 늘림

			if (num == n) {
				cnt++;
				num--;
			}

			i++;
		}
		else num = 0;
	}

	return cnt;
}

int main() {
	int N; cin >> N;	// PN
	int M; cin >> M;	// 문자열 길이
	string S; cin >> S;	// 문자열

	cout << PN(S, N) << endl;
}

File: 5/05525.py
def PN(s, n):
    cnt = 0
    num = 0

    i = 0
    while(True):
        if i >= len(s) - 2: break

        if s[i] == 'I' and s[i+1] == 'O' and s[i+2] == 'I':
            num += 1

            if num == n:
                cnt += 1
                num -= 1

            i += 1
        else: num = 0

        i += 1

    return cnt

if __name__ == "__main__":
    N = int(input())
    M = int(input())
    S = input()

    print(PN(S, N))
